Library/Formula/postgresql.rb

```diff
@@ -77,8 +82,42 @@ def install
   end
 
   def post_install
-    unless File.exist? "#{var}/postgres"
-      system "#{bin}/initdb", "#{var}/postgres"
+    if File.exist? "#{var}/postgres/PG_VERSION"
+      pg_version = (var/"postgres/PG_VERSION").read.chomp
+    end
+
+    if (var/"postgres").exist? && (var/"postgres-#{pg_version}").exist?
+      puts "We have not attempted to migrate your data due to your multi-postgresql setup."
+    end
+
+    # With the 9.5 release, we'll need to adapt logic to make sure we preserve the
+    # 9.4 var directories, without breaking postgres. This should be able to be done
+    # by modifying some lines below to var/"postgres-4.9" instead of var/"postgres"
+    old_path = var/"postgres"
+    new_path = var/"postgres-#{version_suffix}"
+    # This should always be the last full version shipped by Brew, i.e. 9.4.0.
+    prior_ver = "9.4.0"
```

Comment:
But ` PG_VERSION ` won't match up with the Cellar version, because ` PG_VERSION ` only covers x.x whereas our Cellar is x.x.x? Unless I'm wildly misunderstanding you? Which seems likely :grin: